### PR TITLE
Additional testing of status concept of the board and additional documentation for the already existing test cases for the board.

### PR DESCRIPTION
**Pull request by tim-gr**

### CODE REVIEW

File: test/Board.cpp [hunk 1]
```diff
@@ -77,3 +92,154 @@ TEST_CASE("Mixed INO/C++ sources", "[Board]") {
...
+TEST_CASE("Board - Check conditions for configure", "[Board]") {
+    smce::Toolchain tc{SMCE_PATH};
+    REQUIRE(!tc.check_suitable_environment());
+
+    smce::Sketch sk{SKETCHES_PATH "noop", {.fqbn = "arduino:avr:nano"}};
+    tc.compile(sk);
+    REQUIRE(sk.is_compiled());
+    smce::Board br{};
+
+    // Board is both clean and configured => Configuration possible
```
Comment (BerglundLukas) [inline test/Board.cpp:158]: In the comment, you say "and configured" for the board. Shouldn't it be "and not configured"?

File: test/Board.cpp [hunk 2]
```diff
@@ -77,3 +92,154 @@ TEST_CASE("Mixed INO/C++ sources", "[Board]") {
...
+    REQUIRE(br.stop());
+    REQUIRE(br.status() == smce::Board::Status::stopped);
+    REQUIRE(br.attach_sketch(sk2));
+    REQUIRE(br.reset());
+}
+
+/**
+ * Test that configuring the board only works
+ * when the board is clean or already configured.
+ */
```
Comment (BerglundLukas) [inline test/Board.cpp:148]: Does "Clean" mean that the board is not yet configured? I.e it is possible to configure the board if the board is not yet configured or has already been configured?

File: test/Board.cpp (hunk 1 above)
Comment (tim-gr) [reply test/Board.cpp:158]: Absolutely, thanks :D

File: test/Board.cpp (hunk 2 above)
Comment (tim-gr) [reply test/Board.cpp:148]: It means that you can only configure the board, if:

- The board has the status configured, so the first configuration was done just before.
- The board is clean, so in the initial state. For example, a board that has already run, cannot be configured again. Instead, it has to be cleaned/reset first.

Comment (tim-gr) [reply test/Board.cpp:148]: I will improve the comment in the test case.

Comment (BerglundLukas) [reply test/Board.cpp:148]: Looks good now

**Review by BerglundLukas — APPROVED** [review]

**Pull request merged**